tdms/src/iterator.cpp

```
@@ -371,1006 +319,577 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
-          place_holder = mxGetPr( (mxArray *)element);
+          *(I0 + 1) = *(place_holder + 1);
+        } else if (are_equal(interface_fields[i], "I1")) {
+          place_holder = mxGetPr((mxArray *) element);
           *I1 = *place_holder - 1.;
-          *(I1+1) = *(place_holder+1);
-        }
-        else if(!strcmp(interface_fields[i],"J0")){
-          place_holder = mxGetPr( (mxArray *)element);
+          *(I1 + 1) = *(place_holder + 1);
+        } else if (are_equal(interface_fields[i], "J0")) {
+          place_holder = mxGetPr((mxArray *) element);
           *J0 = *place_holder - 1.;
-          *(J0+1) = *(place_holder+1);
-        }
-        else if(!strcmp(interface_fields[i],"J1")){
-          place_holder = mxGetPr( (mxArray *)element);
+          *(J0 + 1) = *(place_holder + 1);
+        } else if (are_equal(interface_fields[i], "J1")) {
+          place_holder = mxGetPr((mxArray *) element);
           *J1 = *place_holder - 1.;
-          *(J1+1) = *(place_holder+1);
-        }
-        else if(!strcmp(interface_fields[i],"K0")){
-          place_holder = mxGetPr( (mxArray *)element);
+          *(J1 + 1) = *(place_holder + 1);
+        } else if (are_equal(interface_fields[i], "K0")) {
+          place_holder = mxGetPr((mxArray *) element);
           *K0 = *place_holder - 1.;
-          if(*K0<0)
-            *K0 = 0.;
-          *(K0+1) = *(place_holder+1);
-        }
-        else if(!strcmp(interface_fields[i],"K1")){
-          place_holder = mxGetPr( (mxArray *)element);
+          if (*K0 < 0) *K0 = 0.;
+          *(K0 + 1) = *(place_holder + 1);
+        } else if (are_equal(interface_fields[i], "K1")) {
+          place_holder = mxGetPr((mxArray *) element);
           *K1 = *place_holder - 1.;
-          if(*K1<0)
-            *K1 = 0.;
-          *(K1+1) = *(place_holder+1);
-        }
-        else{
-          throw runtime_error("element interface."+field_name+" not handled");
+          if (*K1 < 0) *K1 = 0.;

```

This silently ignores if K is not a viable index in the input and sets it to zero. Maybe something should be thrown here?